python/hn3l_cfg.py

```diff
@@ -0,0 +1,422 @@
+        useLumiInfo=False
+    )
+
+    jsonAna = cfg.Analyzer(
+        JSONAnalyzer,
+        name='JSONAnalyzer',
+    )
+
+    skimAna = cfg.Analyzer(
+        SkimAnalyzerCount,
+        name='SkimAnalyzerCount'
+    )
+
+    triggerAna = cfg.Analyzer(
+        TriggerAnalyzer,
+        name='TriggerAnalyzer',
+        addTriggerObjects=True,
+        requireTrigger=True,
+        usePrescaled=False,
+        unpackLabels=True,
+    )
+
+    vertexAna = cfg.Analyzer(
+        VertexAnalyzer,
+        name='VertexAnalyzer',
+        fixedWeight=1,
+        keepFailingEvents=False,
+        verbose=False
+    )
+
+    pileUpAna = cfg.Analyzer(
+        PileUpAnalyzer,
+        name='PileUpAnalyzer',
+        true=True
+    )
+
+    metFilter = cfg.Analyzer(
+        METFilter,
+        name='METFilter',
+        processName='PAT', #mschoene: Filters very much do exist in MC and most of them should be applied to MC as well, but not all!
+        fallbackProcessName = 'RECO',
+        triggers=[
+            'Flag_goodVertices',
+            'Flag_globalSuperTightHalo2016Filter',
+            'Flag_HBHENoiseFilter',
+            'Flag_HBHENoiseIsoFilter',
+            'Flag_EcalDeadCellTriggerPrimitiveFilter',
+            'Flag_BadPFMuonFilter',
+            'Flag_BadChargedCandidateFilter',
+            'Flag_eeBadScFilter',
+            'Flag_ecalBadCalibFilter',
+        ]
+    )
+
+    HNLTreeProducer = cfg.Analyzer(
+        HNLTreeProducer,
+        name='HNLTreeProducer',
+        L1L2LeptonType=L1L2LeptonType,
+        promptLepType=promptLeptonType,
+    )
+
+    HNLGenTreeAnalyzer = cfg.Analyzer(
+        HNLGenTreeAnalyzer,
+        name='HNLGenTreeAnalyzer',
+    )
+
+    RecoGenAnalyzer = cfg.Analyzer(
+        RecoGenAnalyzer,
+        name='RecoGenAnalyzer',
+    )
+
+    genAna = GeneratorAnalyzer.defaultConfig
+    genAna.allGenTaus = True # save in event.gentaus *ALL* taus, regardless whether hadronic / leptonic decay
+
+    # for each path specify which filters you want the electrons/muons to match to
+    triggers_and_filters = OrderedDict()
+
+    if promptLeptonType == 'ele':
+        triggers_and_filters['HLT_Ele27_WPTight_Gsf']         = 'hltEle27WPTightGsfTrackIsoFilter'
+        triggers_and_filters['HLT_Ele32_WPTight_Gsf']         = 'hltEle32WPTightGsfTrackIsoFilter'
+        triggers_and_filters['HLT_Ele35_WPTight_Gsf']         = 'hltEle35noerWPTightGsfTrackIsoFilter'
+        triggers_and_filters['HLT_Ele115_CaloIdVT_GsfTrkIdT'] = 'hltEle115CaloIdVTGsfTrkIdTGsfDphiFilter'
+        triggers_and_filters['HLT_Ele135_CaloIdVT_GsfTrkIdT'] = 'hltEle135CaloIdVTGsfTrkIdTGsfDphiFilter'
+
+    if promptLeptonType == 'mu':
+        triggers_and_filters['HLT_IsoMu24'] = 'hltL3crIsoL1sSingleMu22L1f0L2f10QL3f24QL3trkIsoFiltered0p07'
+        triggers_and_filters['HLT_IsoMu27'] = 'hltL3crIsoL1sMu22Or25L1f0L2f10QL3f27QL3trkIsoFiltered0p07'
+        triggers_and_filters['HLT_Mu50']    = 'hltL3fL1sMu22Or25L1f0L2f10QL3Filtered50Q'
```

Comment:
you'll need to find the corresponding trigger filters too

Reply:
are you going to add them?

Reply:
Ok, we have discussed now that this trigger collection is complete for 2017 data, the additional ones we will need for 2016 data.

Reply:
2017 triggers are ok, this is settled, no need to add anything